pkg/sharing/namespace_wildcard_exclusions.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package sharing
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// makeNamespaceWildcardExclusionCheck returns a function that uses reconciler-level tools (k8s client, logger, context) to
+// check the presence of a namespace annotation that we mostly only care about in the inner workings of SecretExport.
+func makeNamespaceWildcardExclusionCheck(ctx context.Context,
+	kubernetesClient client.Client,
+	log logr.Logger) NamespaceWildcardExclusionCheck {
+	return func(nsName string) bool {
+		query := types.NamespacedName{
+			Name: nsName,
+		}
+		namespace := corev1.Namespace{}
+		err := kubernetesClient.Get(ctx, query, &namespace)
+		if err != nil {
+			log.Error(err, "Called to check annotation on a namespace but couldn't find:", "namespace", nsName)
```

Comment:
Is there a `log.Errorf` func equivalent for this?

Reply:
i think log package is meant to capture KVs instead of usual sprintf formatting

Reply:
i'm not sure what you're getting at:   you don't like the "structured" part of the logging?

Reply:
I think the option is "call fmt.sprintf (or similar) for yourself if you really want a formatted string"